### PR TITLE
add interactor for profiles controller

### PR DESCRIPTION
**Pull request by tyranja**
In order to make the code in the controller more readable, I started to use interactors.

I used the interactors to encapsulate logic and move it out of the controller. This has the advantage that it can also be  tested separately.

https://github.com/collectiveidea/interactor
https://medium.com/@lucaspolonuer/ruby-on-rails-boost-your-software-design-skills-with-interactors-e07f1f9a711e
https://blog.saeloun.com/2020/05/06/rails-using-interactor-gem/

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -68,175 +63,106 @@ def render_footer?
...
 
-  # Use callbacks to share common setup or constraints between actions.
   def set_profile
     @profile = Profile.friendly.find(params[:id])
   end
 
-  # Never trust parameters from the scary internet, only allow the white list through.
   def profile_params
     params.require(:profile).permit(
-      :email,
```
Comment (zaziemo) [inline app/controllers/profiles_controller.rb:88]: why did you organize the params like this? I found it more readable before.

File: app/controllers/profiles_controller.rb [hunk 2]
```diff
@@ -5,9 +5,9 @@ class ProfilesController < ApplicationController
   include CategoriesHelper
   include ActsAsTaggableOn::TagsHelper
   include SearchHelper
+  include Pagy::Backend
```
Comment (zaziemo) [inline app/controllers/profiles_controller.rb:8]: why did you add this inclusion? Was it missing before or did you change anything else with pagy which I did not see?

File: app/controllers/profiles_controller.rb [hunk 3]
```diff
@@ -68,175 +63,106 @@ def render_footer?
...
-      chain = chain.by_country(params[:filter_country])
-    end
-
-    if params[:filter_language]
-      chain = chain.by_language(params[:filter_language])
-    end
-
-    if params[:filter_state]
-      chain = chain.by_state(params[:filter_state])
+    result = SearchProfiles.call(params: params, region: current_region)
```
Comment (zaziemo) [inline app/controllers/profiles_controller.rb:89]: Probably we should find a better name to distinguish the interactors better. I only can come up with `SearchProfileByAllParams` to describe better what really happens in contrast to `SearchProfileByTag`etc.

File: app/interactor/default_search_profiles.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class DefaultSearchProfiles
+  include Interactor
+
+  def call
+    context.profiles = profiles_for_index
+  end
+
+  private
+
+
```
Comment (zaziemo) [inline app/interactor/default_search_profiles.rb:10]: remove one empty line

**Review by zaziemo — CHANGES_REQUESTED** [review]
I like the interactors and did not know them before :) Let's try it out!
Everything seems to work, I only have some minor improvements.

File: app/controllers/profiles_controller.rb (hunk 1 above)
Comment (tyranja) [reply app/controllers/profiles_controller.rb:88]: It is was more readable before, but also increases the page size a lot in length.

File: app/controllers/profiles_controller.rb (hunk 2 above)
Comment (tyranja) [reply app/controllers/profiles_controller.rb:8]: with this I can use the pagy method: _pagy_array_

**Review by zaziemo — APPROVED** [review]
good work :)

File: spec/interactor/search_profiles_by_params_spec.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  let(:context) { { params: params } }
+  let!(:profile) { create(:profile, city: 'Elend', published: true, main_topic: "Handarbeit") }
+
+  before do
+    create(:profile, main_topic: "Handarbeit", published: false )
+    create(:profile, main_topic: "Baumschnitt", published: true )
+  end
+
+  it 'only shows published profiles with the correct params' do
+    byebug
```
Comment (zaziemo) [inline spec/interactor/search_profiles_by_params_spec.rb:12]: delete byebug :)
